### PR TITLE
Cache all memberNamed results

### PR DESCRIPTION
**Pull request by odersky**
Use a full cache instead of an LRU cache. For `typer/*.scala`, this
reduced computed member searches on normal ClassDenotations from 520K to
170K. Cache hit rate improves to 97.5%, from 92.5%. (Without member caching
there are almost 7Mio computed members for the same code base).

### CODE REVIEW

**Comment by odersky** [comment]
Back of the envelope calculation for memory sizes. All numbers for typer/*.scala with Stats enabled, so numbers are a bit higher than without.

We have 174K computeNonPrivateMembersNamed

Caches: 4657 ArrayTables
               1293 HashTables

Small TableSizes:

size #tables

  0  1532 
  1   1689
  2   564
  3   366
  4   200
  5   175
  6   120
  7   121
  8   90

Total entries in ArrayTables: 13788. This leaves ~160K entries in HashTables. So member accesses are spread out very unevenly. The majority is in relatively few large tables. 

Largest 5 table sizes: 4079, 4045, 2353, 1443, 1400. Then there are 20 or so between 1000 and 1300. I suspect the two
largest sizes are for Any and Object.

Size overhead: ArrayTables: 4657 * 88 bytes = 410K, or 30 bytes per used entry
Size overhead: HashTables: approx. 30bytes/entry --> ~4.8M
Total ~5.2M

Previous solution with LRU caches:

5950 caches * 176 bytes ~ 1M

Bottom line: we get a 3x improvement in cache hit rate (from 92.5% to 97.5%) for a 5x increase in memory (1M -> 5.2M).

**Comment by odersky** [comment]
test performance please

**Comment by odersky** [comment]
Downstream improvement: # allocated AsSeenFromMap goes from 124572 to 93616

**Comment by odersky** [comment]
test performance please

**Review by liufengyun — APPROVED** [review]
LGTM

**Comment by odersky** [comment]
test performance please

**Comment by odersky** [comment]
@nicolasstucki Can you help me debug the idempotency failure here? In particular, is there a way to repeoduce this in a standalone test? I cannot reproduce the failure locally, it's CI only.

**Comment by odersky** [comment]
The recent deterioration is due to parallel position pickling being disabled now.
